Mathlib/Analysis/Discrete/MarcinkiewiczZygmund.lean

```diff
@@ -0,0 +1,286 @@
+/-
+Copyright (c) 2023 Yaël Dillies, Bhavik Mehta. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yaël Dillies, Bhavik Mehta
+-/
+import Mathlib.Analysis.InnerProductSpace.PiL2
+import Mathlib.Analysis.MeanInequalitiesPow
+import Mathlib.Data.Nat.Choose.Multinomial
+import Mathlib.Tactic.Positivity.Finset
+
+/-!
+# The Marcinkiewicz-Zygmund inequality
+
+This file proves the Marcinkiewicz-Zygmund inequality. This is the statement that, for a random
```

Comment:
I am really surprised by this version: for me, one of the important points of Marcinkiewicz-Zygmund is that you can take random variables which are not identically distributed. Also, you are working with the specific case of a fintype, while the inequality in true for a general probability space. I'd rather deduce the particular version from the general case (which is not really harder, although I'd rather get the more precise Rosenthal or Burkholder inequalities), but if you want to focus on the discrete space you should emphasize that in the file-level docstring.

Reply:
Indeed this is a special case, but it's precisely the special case which arises as a very early dependence in our work. Would you mind if we stayed with this version for now, with a comment to generalise it in the way you describe?

Reply:
I absolutely don't mind, as long as it's explained clearly in the file-level docstring.

Reply:
Tried adding something